docs/docs/porting-create-react-app-to-gatsby.md

```diff
@@ -0,0 +1,263 @@
+
+> **Note**: You could import your top level `App.js` in a file at `/src/pages/index.js` and your project will likely run with the `gatsby develop` command already, though you would lose many of the benefits like code splitting and server-side rendering that Gatsby uses for pages
+
+### Server-side rendering and browser APIs
+
+Server-side rendering means pages and dynamic data are built out by the server, and then sent to a browser ready to go. It's like your page is loaded before even being sent to the user. Gatsby is server-side rendered at build time, meaning that the code that gets to your browser has already been run to build pages and content, but this doesn't mean you can't still have dynamic pages.
+
+Understanding the distinction between the client (or browser) and server will help you understand that key difference between Create React App and Gatsby. Create React App doesn't by default render your components with server-side rendering APIs when it is built like Gatsby does.
+
+_Check out Dustin Schau's [blog post about Gatsby internals](/blog/2019-04-02-behind-the-scenes-what-makes-gatsby-great/#server-side-rendering-ssr-at-build-time) that explains the technical aspects of the build process in greater detail_
+
+The `gatsby build` command won't be able to use browser APIs so some code would cause your build to break if it isn't protected.
+
+Some common APIs that would need to be protected are:
+
+- `window`
+- `localStorage`
+- `navigator`
+- packages like `react-router-dom`
+
+These are only a few examples, though all can be fixed in one of two ways:
+
+1. wrapping the code in an `if` to check if whatever you are referencing is defined so builds won't try to reference something undefined and the browser will be able to deal with it fine:
+
+```jsx
+if (typeof window !== `undefined`) {
+  // code using window like window.location...
+}
+```
+
+2. moving references to them into a `componentDidMount` or `useEffect` hook:
+
+```jsx
+import React from "react"
+
+const Foo = () => {
+  window.alert("This will break the build")
+  return <span>Bar</span>
+}
+
+export default Foo
+```
+
+Would be changed to:
+
+```jsx
+import React from "react"
+
+const Foo = () => {
+  React.useEffect(() => {
+    window.alert("This won't break the build")
+  })
+  return <span>Bar</span>
+}
+
+export default Foo
+```
+
+For more information about errors encountered during builds, see the doc on [debugging HTML builds](/docs/debugging-html-builds/)
+
+### Routing
+
+There are two possibilites of routes that you can setup: static and dynamic. A static route has content that doesn't change, whereas a dynamic route can [fetch data at runtime](/docs/client-data-fetching/) just like any other React app.
```

Comment:
```suggestion
There are two possibilites of routes that you can set up: static and dynamic. A static route has content that doesn't change, whereas a dynamic route can [fetch data at runtime](/docs/client-data-fetching/) just like any other React app.
```
This is a bit of a nitpick, I know...but differentiating from the actionable phrase "set up" as in "to set up" from "setup" as a noun.